doc/subsystems/nvs/nvs.rst

```diff
@@ -0,0 +1,112 @@
+.. _nvs:
+
+Non-volatile storage (NVS) for Zephyr
+#####################################
+
+Elements, represented as id-data pairs, are stored in flash using a
+FIFO-managed circular buffer. The flash area is divided into sectors. Elements
+are appended to a sector until storage space in the sector is exhausted. Then a
+new sector in the flash area is prepared for use (erased). Before erasing the
+sector it is checked that identifier - data pairs exist in the sectors in use,
+if not the id-data pair is copied.
+
+The id is a 16-bit unsigned number, where two values are reserved:
+
+- ``0xFFFF`` is used to determine free locations in flash
+- ``0xFFFE`` is used to close a sector
+
+NVS ensures that for each id there is at least one id-data pair stored in flash
+at all time.
+
+NVS allows storage of binary blobs, strings, integers, longs, and any
+combination of these. As each element is stored with a header (containing the
+id and length) and a footer (containing the crc16) it is less suited to store
+only integers.
+
+NVS has a configuration option to enable extra flash protection
+(``CONFIG_NVS_FLASH_PROTECTION=y``), when selected NVS does a extra check
+before writing data to flash. If the id-data pair is unchanged no write to
+flash is performed. The down-side of this protection is that NVS needs to read
+the storage system to check if the id-data pair is unchanged. If you are
+already performing such a check you can disable the extra flash protection
+(``CONFIG_NVS_FLASH_PROTECTION=n``).
+
+To protect the storage system against frequent sector erases the size of
+id-data pairs is limited to ``CONFIG_NVS_MAX_ELEM_SIZE``. This limit is by default set to 1/4 of the sector
+size.
+
+
+For NVS the file system is declared as:
+
+.. code-block:: c
+
+	static struct nvs_fs fs = {
+	.sector_size = CONFIG_NVS_SECTOR_SIZE,
+	.sector_count = CONFIG_NVS_SECTOR_COUNT,
+	.offset = FLASH_AREA_STORAGE_OFFSET,
+	.max_len = CONFIG_NVS_MAX_ELEM_SIZE,
+	};
+
+where
+
+- ``CONFIG_NVS_SECTOR_SIZE`` is the sector size, it has to be a multiple of
+the flash erase page size and a power of 2.
```

Comment:
The continuation line of a list needs to be indented to the first non-white space of the line above it.  So change this (and the lines below) to add a couple spaces on the continuation lines, like so:

    - ``CONFIG_NVS_SECTOR_SIZE`` is the sector size, it has to be a multiple of
      the flash erase page size and a power of 2.